core/src/main/java/de/valtech/aecu/core/groovy/console/bindings/actions/resource/AddMixin.java

```diff
@@ -0,0 +1,49 @@
+package de.valtech.aecu.core.groovy.console.bindings.actions.resource;
+
+import de.valtech.aecu.api.service.AecuException;
+import de.valtech.aecu.core.groovy.console.bindings.actions.Action;
+import de.valtech.aecu.core.groovy.console.bindings.actions.util.MixinUtil;
+import javax.annotation.Nonnull;
+import javax.jcr.Node;
+import javax.jcr.RepositoryException;
+import javax.jcr.nodetype.NoSuchNodeTypeException;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.resource.PersistenceException;
+import org.apache.sling.api.resource.Resource;
+
+/**
+ * Adds a mixin to a resource/node if the mixin exists.
+ *
+ * @author Eric Manzi and Bart Thierens
+ */
+public class AddMixin implements Action {
+
+    private final String mixinName;
+
+    public AddMixin(String mixinName) {
+        this.mixinName = mixinName;
+    }
+
+    @Override
+    public String doAction(@Nonnull Resource resource) throws PersistenceException {
+        if (StringUtils.isBlank(mixinName)) {
+            return "WARNING: mixin name is empty";
+        }
+
+        Node node = resource.adaptTo(Node.class);
+        if (node == null) {
+            return "WARNING: could not get node for " + resource.getPath();
+        }
+
+        try {
```

Comment:
the `context.isDryRun()` logic would need to be implemented.

Reply:
@nhirrle why is that?  I just verified it to be sure, and when you do a dry-run the session isn't saved at the end of the content-upgrade and the mixins aren't added.  It does print out "Adding mixin mix:versionable to /content/dam/jcr:content".  So that seems to work correctly?

Other existing action like SetProperty also don't have a specific dryRun-impl.

Reply:
You are right @senn , sorry for that. I had a look on SetPageTagsAction but here we are autosaving the session. Sor please ignore my comment